src/views/Dashboard/Customer/Home/RequestTest/ServiceDetailDialog.vue

```diff
@@ -119,15 +123,20 @@ export default {
     },
 
     async downloadFile () {
-      const publicKey = hexToU8a(this.mnemonicData.publicKey)
-      const privateKey = hexToU8a(this.mnemonicData.privateKey)
+
+      const cred = Kilt.Identity.buildFromMnemonic(this.mnemonicData.toString(CryptoJS.enc.Utf8))
+
+      const publicKey = u8aToHex(cred.boxKeyPair.publicKey)
+      const privateKey = u8aToHex(cred.boxKeyPair.publicKey)
       const baseUrl = "https://ipfs.io/ipfs/"
-      const path = this.downloadPath.replace(baseUrl, "")
+      const path = this.selectedService.resultSample.replace(baseUrl, "")
```

Comment:
Please consider using `split` then use `at` if possible. Hardcoding URLs is not a good practice, and AFAIK the last segment of any IPFS URL will be the hash.
- https://www.w3schools.com/jsref/jsref_split.asp
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at